File: src/frontend/src/models/StudentEnrollmentDto.ts
import PersonDto from './PersonDto'

export default class StudentEnrollmentDto {
  id?: number
  student?: PersonDto
  status?: string // "ENROLLED", "APPROVED", "FAILED"
  enrollmentDate?: string
  finalGrade?: number

  constructor(obj?: Partial<StudentEnrollmentDto>) {
    Object.assign(this, obj)
  }
}

File: src/backend/src/main/java/pt/ulisboa/tecnico/rnl/dei/dms/studentEnrollment/dto/StudentEnrollmentDto.java
package pt.ulisboa.tecnico.rnl.dei.dms.studentEnrollment.dto;

import pt.ulisboa.tecnico.rnl.dei.dms.person.dto.PersonDto;
import pt.ulisboa.tecnico.rnl.dei.dms.studentEnrollment.domain.StudentEnrollment;

import java.time.LocalDate;

public record StudentEnrollmentDto(
    Long id,
    PersonDto student,
    String status,
    LocalDate enrollmentDate,
    LocalDate completionDate,
    Double finalGrade
) {
    public StudentEnrollmentDto(StudentEnrollment enrollment) {
        this(
            enrollment.getId(),
            new PersonDto(enrollment.getStudent()),
            enrollment.getStatus().toString(),
            enrollment.getEnrollmentDate(),
            enrollment.getCompletionDate(),
            enrollment.getFinalGrade()
        );
    }
}
